From 1f6937eb336a6d1bd0dfaf3fe5d04df5390775e4 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 31 Mar 2022 15:00:35 +0000
Subject: [PATCH] libstdc++: fix incomplete type of ModelProvider

Destructor of std::unique_ptr in libstdc++ uses sizeof() which
requires full definition of segmentation_platform::ModelProvider
in segmentation_platform::GetSegmentationDefaultModelProvider().
---
 .../segmentation_platform/segmentation_platform_config.cc        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/chrome/browser/segmentation_platform/segmentation_platform_config.cc b/chrome/browser/segmentation_platform/segmentation_platform_config.cc
index b2bf48e..1c817ae 100644
--- a/chrome/browser/segmentation_platform/segmentation_platform_config.cc
+++ b/chrome/browser/segmentation_platform/segmentation_platform_config.cc
@@ -12,6 +12,7 @@
 #include "components/optimization_guide/proto/models.pb.h"
 #include "components/segmentation_platform/public/config.h"
 #include "components/segmentation_platform/public/features.h"
+#include "components/segmentation_platform/public/model_provider.h"
 
 #if BUILDFLAG(IS_ANDROID)
 #include "base/metrics/field_trial_params.h"
-- 
2.34.1

